src/main/java/organice/logic/commands/MatchCommand.java

```diff
@@ -0,0 +1,93 @@
+package organice.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Predicate;
+
+import organice.model.Model;
+import organice.model.person.BloodType;
+import organice.model.person.Donor;
+import organice.model.person.Nric;
+import organice.model.person.Patient;
+import organice.model.person.Person;
+import organice.model.person.TissueType;
+import organice.model.person.exceptions.PersonNotFoundException;
+
+/**
+ * Lists all persons in the address book to the user.
+ */
+public class MatchCommand extends Command {
+
+    public static final String COMMAND_WORD = "match";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Matches patients and donors who passes matching tests"
+            + "Parameters: ic/all (to match all patients and donors)"
```

Comment:
I feel that having /n for this part will be nicer since user do not need to scroll to see everything :)